### PR TITLE
[Sonos]: add an mvp for a sonos extension

### PR DESCRIPTION
**Pull request by AntonNiklasson**
## Description

I've been trying to wrap my head around the Sonos API, and how to integrate it into Raycast. I think this would be a decent starting point to just get something out there. I'm excited to keep working on it, and add more features.

I've added the following commands:

- Play/pause toggle
- Set active group: for when multiple groups are available in the sonos system
- Now playing: a menu bar command to simply show what's currently playing
- Increase + Decrease volume: currently hard coded to change the volume in steps of 2. This might be a user preference at some point
- Next + Previous track

Screenshot 2024-01-30 at 13 59 47

Screen Recording 2024-01-07 at 21 28 09

### CODE REVIEW

**Comment by AntonNiklasson** [comment]
I'll get back to this soon 🙌

**Comment by AntonNiklasson** [comment]
@pernielsentikaer I'm ready to merge this, if you're up for it 🙌

**Comment by thomaspaulmann** [comment]
Oh yes, I've been waiting for a Sonos extension for a while, can't wait!

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- Could you look into handling errors like this gracefully?

sonos 2024-02-02 at 12 55 02

- Could you add at least one [screenshot](https://developers.raycast.com/basics/prepare-an-extension-for-store#screenshots) of the extension

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/sonos/package.json [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      "subtitle": "Sonos",
+      "disabledByDefault": true,
+      "description": "Increase volume in the active group",
+      "mode": "no-view"
+    },
+    {
+      "name": "decrease-volume",
+      "title": "Decrease Volume",
+      "subtitle": "Sonos",
+      "disabledByDefault": true,
```
Comment (AntonNiklasson) [inline extensions/sonos/package.json:49]: What was the reason for this change, @pernielsentikaer? If we handle the case of running these commands without a Sonos system present, perhaps all commands can be enabled by default? Or do you have some best-practice in mind around this?

Comment (pernielsentikaer) [reply extensions/sonos/package.json:49]: Replied back on your Slack DM 🙂

Comment (AntonNiklasson) [reply extensions/sonos/package.json:49]: I've disabled the skip commands by default as well 🙌

**Comment by AntonNiklasson** [comment]
> * Could you look into handling errors like this gracefully?

I've improved the error handling to handle the "missing Sonos" situation a bit better. The commands are now showing a failure toast instead:

Raycast 2024-02-05 at 22 44 05

File: extensions/sonos/src/decrease-volume.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  } catch (error) {
+    await handleCommandError(error);
+    return;
+  }
+
+  if (coordinator === undefined) {
+    await launchCommand({
+      name: "set-group",
+      type: LaunchType.UserInitiated,
+    });
```
Comment (pernielsentikaer) [inline extensions/sonos/src/decrease-volume.tsx:20]: You need to have try catch around launchCommand too (applies to all commands) 🙂

Comment (AntonNiklasson) [reply extensions/sonos/src/decrease-volume.tsx]: Ah, nice catch! I've introduced `tryLaunchCommand` and use it for all those calls now.

**Pull request merged**